go/arrow/array/concat_test.go

```diff
@@ -289,13 +288,3 @@ func (cts *ConcatTestSuite) TestCheckConcat() {
 		})
 	}
 }
-
-func TestOffsetOverflow(t *testing.T) {
-	fakeOffsets := memory.NewBufferBytes(arrow.Int32Traits.CastToBytes([]int32{0, math.MaxInt32}))
-	fakeArr := array.NewStringData(array.NewData(arrow.BinaryTypes.String, 1, []*memory.Buffer{nil, fakeOffsets, memory.NewBufferBytes([]byte{})}, nil, 0, 0))
-	var err error
-	assert.NotPanics(t, func() {
-		_, err = array.Concatenate([]arrow.Array{fakeArr, fakeArr}, memory.DefaultAllocator)
-	})
-	assert.EqualError(t, err, "offset overflow while concatenating arrays")
-}
```

Comment:
This test now panics in `NewStringData` because that calls down to `setData`. Should I replace that with manual string array creation like in `TestStringInvalidOffsets`?

Reply:
Ah gotcha, we're dying there rather than inside `Concatenate` good point. Okay then yea it makes sense to just remove this test since we can't hit this situation anymore with the addition of the panics for the bad offset arrays.